src/mapnik_image.cpp

```diff
@@ -1421,6 +1410,460 @@ Local<Value> Image::_copySync(_NAN_METHOD_ARGS)
+    }    
+    if (args.Length() >= 3)
+    {
+        if (args[2]->IsObject())
+        {
+            options = args[2]->ToObject();
+        }
+        else
+        {
+            NanThrowTypeError("Expected options object as third argument");
+            return NanEscapeScope(NanUndefined());
+        }
+    }
+    
+    if (options->Has(NanNew("scaling_method")))
+    {
+        Local<Value> scaling_val = options->Get(NanNew("scaling_method"));
+        if (scaling_val->IsNumber())
+        {
+            scaling_method = static_cast<mapnik::scaling_method_e>(scaling_val->IntegerValue());
+            if (scaling_method > mapnik::SCALING_BLACKMAN)
+            {
+                NanThrowTypeError("Invalid scaling_method");
+                return NanEscapeScope(NanUndefined());
+            }
+        }
+        else
+        {
+            NanThrowTypeError("scaling_method argument must be a number");
+            return NanEscapeScope(NanUndefined());
+        }
+    }
+    
+    if (options->Has(NanNew("filter_factor")))
+    {
+        Local<Value> ff_val = options->Get(NanNew("filter_factor"));
+        if (ff_val->IsNumber())
+        {
+            filter_factor = ff_val->NumberValue();
+        }
+        else
+        {
+            NanThrowTypeError("filter_factor argument must be a number");
+            return NanEscapeScope(NanUndefined());
+        }
+    }
+
+    if (im->this_->is<mapnik::image_null>())
+    {
+        NanThrowTypeError("Can not resize a null image");
+        return NanEscapeScope(NanUndefined());
+    }
+    int im_width = im->this_->width();
+    int im_height = im->this_->height();
+    if (im_width <= 0 || im_height <= 0)
+    {
+        NanThrowTypeError("Image width or height is zero or less then zero.");
+        return NanEscapeScope(NanUndefined());
+    }
+    try
+    {
+        double offset = im->this_->get_offset();
+        double scaling = im->this_->get_scaling();
+
+        std::shared_ptr<mapnik::image_any> image_ptr = std::make_shared<mapnik::image_any>(width, 
+                                                           height, 
+                                                           im->this_->get_dtype(),
+                                                           false,
+                                                           true,
+                                                           false);
+        image_ptr->set_offset(offset);
+        image_ptr->set_scaling(scaling);
+        double image_ratio_x = static_cast<double>(width) / im_width; 
+        double image_ratio_y = static_cast<double>(height) / im_height; 
+        resize_visitor visit(*(im->this_),
+                             scaling_method,
+                             image_ratio_x,
+                             image_ratio_y,
+                             filter_factor);
+        mapnik::util::apply_visitor(visit, *image_ptr);
+        Image* new_im = new Image(image_ptr);
```

Comment:
No coverage for this line. So no proper testing with actual image returned from Sync usage. Add tests.